src/test/java/com/artipie/pypi/PypiITCase.java

```diff
@@ -23,205 +23,85 @@
-        this.init(anonymous);
-        new TestResource("pypi-repo/alarmtime-0.1.5.tar.gz")
-            .saveTo(
-                this.storage,
-                new Key.From("repos", "my-pypi", "alarmtime", "alarmtime-0.1.5.tar.gz")
-        );
-        MatcherAssert.assertThat(
-            this.cntn.execStdout(
-                "pip", "install", "--no-deps", "--trusted-host", PypiITCase.HOST,
-                "--index-url", this.url.string(anonymous),
-                "alarmtime==0.1.5"
-            ),
-            new StringContains("Successfully installed alarmtime-0.1.5")
-        );
-    }
-
-    @ParameterizedTest
-    @ValueSource(booleans = {true, false})
-    void pypiPublish(final boolean anonymous) throws Exception {
-        this.init(anonymous);
-        new TestResource("pypi-repo/example-pckg/dist/")
-            .addFilesTo(
-                this.storage,
-                new Key.From("pypi-repo", "example-pckg")
-        );
-        this.cntn.execStdout("python3", "-m", "pip", "install", "--user", "--upgrade", "twine");
-        MatcherAssert.assertThat(
-            "Packages should be uploaded",
-            this.cntn.execStdout(
-                "python3", "-m", "twine", "upload", "--repository-url", this.url.string(),
-                "-u", ArtipieServer.ALICE.name(), "-p", this.pswd(anonymous),
-                "pypi-repo/example-pckg/*"
-            ),
-            new StringContainsInOrder(
-                new ListOf<String>(
-                    "Uploading artipietestpkg-0.0.3-py2-none-any.whl", "100%",
-                    "Uploading artipietestpkg-0.0.3.tar.gz", "100%"
-                )
+    @RegisterExtension
+    final TestDeployment containers = new TestDeployment(
+        () -> TestDeployment.ArtipieContainer.defaultDefinition()
+            .withRepoConfig("pypi-repo/pypi.yml", "my-python"),
+        () -> new TestDeployment.ClientContainer("python")
+            .withWorkingDirectory("/var/artipie")
+            .withClasspathResourceMapping(
+                "pypi-repo/alarmtime-0.1.5.tar.gz",
+                "/var/artipie/data/artipie/pypi/alarmtime/alarmtime-0.1.5.tar.gz",
+                BindMode.READ_ONLY
+            )
+            .withClasspathResourceMapping(
+                "pypi-repo/example-pckg",
+                "/var/artipie/data/artipie/pypi/example-pckg",
+                BindMode.READ_ONLY
             )
+    );
+
+    @BeforeEach
+    void setUp() throws IOException {
+        this.containers.assertExec(
+            "Apt-get update failed",
+            new MavenITCase.ContainerResultMatcher(),
+            "apt-get", "update"
         );
-        MatcherAssert.assertThat(
-            "Packages should be saved in storage",
-            this.inStorage("artipietestpkg-0.0.3-py2-none-any.whl")
-                && this.inStorage("artipietestpkg-0.0.3.tar.gz"),
-            new IsEqual<>(true)
+        this.containers.assertExec(
+            "Install twine",
```

Comment:
@milkysoft assertion message should describe what went wrong and be negative, for example:
Incorrect response status, '200 OK' is expected
Failed to deploy the artifact
String does not match the pattern

Reply:
@olenagerasimova , fixed with latest commit